src/core/segment_sinks/inventory/segment_inventory.ts

```diff
@@ -312,12 +320,28 @@ export default class SegmentInventory {
         let thisSegmentStart = thisSegment.bufferedStart ?? thisSegment.start;
         let thisSegmentEnd = thisSegment.bufferedEnd ?? thisSegment.end;
         const nextRangeStart = i < rangesLength - 1 ? ranges[i + 1].start : undefined;
-        while (
-          thisSegment !== undefined &&
-          rangeEnd - thisSegmentStart >= MINIMUM_SEGMENT_SIZE &&
-          (nextRangeStart === undefined ||
-            rangeEnd - thisSegmentStart >= thisSegmentEnd - nextRangeStart)
-        ) {
+        while (thisSegment !== undefined) {
+          if (rangeEnd < thisSegmentStart) {
+            // `thisSegment` is part of the next range
+            break;
+          }
+          if (
+            rangeEnd - thisSegmentStart < MINIMUM_SEGMENT_SIZE &&
+            thisSegmentEnd - rangeEnd >= MINIMUM_SEGMENT_SIZE
+          ) {
+            // Ambiguous, but `thisSegment` seems more to come after the current
+            // range than during it.
+            break;
+          }
+          if (
+            nextRangeStart !== undefined &&
+            rangeEnd - thisSegmentStart >= thisSegmentEnd - nextRangeStart
```

Comment:
Shouldn't it be the opposite of the previous condition ?

Reply:
I think you're right, I just replaced `>=` by `<`